src/google/protobuf/stubs/strutil.cc

```diff
@@ -503,7 +503,7 @@ int CEscapeInternal(const char* src, int src_len, char* dest,
              (last_hex_escape && isxdigit(*src)))) {
           if (dest_len - used < 4) // need space for 4 letter escape
             return -1;
-          sprintf(dest + used, (use_hex ? "\\x%02x" : "\\%03o"),
+          snprintf(dest + used, 4, (use_hex ? "\\x%02x" : "\\%03o"),
```

Comment:
Some of the tests are failing and I suspect it's because 4 is not enough characters when you include the trailing null byte. So I think this should be a 5 instead of a 4. Fortunately this logic will all be replaced by Abseil fairly soon.

Reply:
Makes sense.  Updated to 5.  Will the tests automatically again?